services-directions-refresh-models/src/main/java/com/mapbox/api/directionsrefresh/v1/models/DirectionsRefreshJsonObject.java

```diff
@@ -25,4 +34,51 @@ public String toJson() {
     gson.registerTypeAdapterFactory(DirectionsRefreshAdapterFactory.create());
     return gson.create().toJson(this);
   }
+
+  /**
+   * Use this method to get names of JSON properties that weren't recognized during JSON
+   * serialization by the model. This may be useful to access experimental API properties
+   * via {@link #getUnrecognizedProperty(String)}.
+   * When an experimental API property becomes stable,
+   * it will eventually have static field in a model introduced
+   * and it won't be available via this dynamic method anymore.
+   *
+   * See
+   * <a href="https://docs.mapbox.com/api/navigation/directions/">Directions API documentation</a>
+   * for available experimental fields.
+   *
+   * @return names of unrecognized JSON properties or an empty set
+   */
+  @NonNull
+  public final Set<String> getUnrecognizedPropertiesNames() {
+    return UnrecognizedPropertiesUtils.getUnrecognizedPropertiesNames(unrecognized());
+  }
+
+  /**
+   * Use this method to access a JSON property that wasn't recognized during JSON serialization
+   * by the model. This may be useful to access experimental API properties.
+   * When an experimental API property becomes stable,
+   * it will eventually have static field in a model introduced
+   * and it won't be available via this dynamic method anymore.
+   *
+   * See
+   * <a href="https://docs.mapbox.com/api/navigation/directions/">Directions API documentation</a>
+   * for available experimental fields.
+   *
+   * @param propertyName name of a json property
+   * @return value of a requested property or null if the requested property doesn't exist.
+   */
+  @Nullable
+  public final JsonElement getUnrecognizedProperty(String propertyName) {
+    return UnrecognizedPropertiesUtils.getUnrecognizedProperty(propertyName, unrecognized());
+  }
+
+  @Nullable
+  @UnrecognizedJsonProperties
+  abstract Map<String, SerializableJsonElement> unrecognized();
```

Comment:
We could consider making this public, refs `NAVAND-778`. Doesn't have to happen in this PR though.

Reply:
How will making this public help `NAVAND-778`? For the case described there it's more like adding a method `addUnrecognizedProperties` instead of the current one that resets them.

Reply:
A public getter would allow us to first get the properties map, mutate them, and set them back. As we would do with any other builder object parameter. The getter and setter methods in this case need to work on the same `map` type so that developers don't need to do any parsing themselves, that's why I was hoping to expose a single function in the abstract class instead of adding explicit getter and setters manually everywhere.

Reply:
OK, makes sense but it's not very convenient to use with `SerializableJsonElement`. But I guess we could add convenience methods directly to `DirectionsJsonObject` and `DirectionsRefreshJsonObject`. WDYT?

Reply:
Sure, as we're discussing in https://github.com/mapbox/mapbox-java/pull/1500#discussion_r1000562709 `JsonElement` might be better.